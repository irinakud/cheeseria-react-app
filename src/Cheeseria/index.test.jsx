import { render, screen, waitFor } from "@testing-library/react";
import { Cheeseria } from ".";

const fakeData = [
    {
      id: '1-1-1',
      type: 'gouda',
      pricePerKilo: 81,
      color: 'yellow',
      pictureFile: 'https://cheese.com/media/img/cheese/GOUDA_SqYJjRh.jpg',
    },
    {
      id: '1-2-1',
      type: 'mozzarella',
      pricePerKilo: 17,
      color: 'white',
      pictureFile: 'https://cheese.com/media/img/cheese/Mozzarella.jpg',
    },
    {
      id: '333',
      type: 'brie',
      pricePerKilo: 100,
      color: 'yellow',
      pictureFile: 'https://cheese.com/media/img/cheese/Brie_PDCo3RG.jpg',
    },
    {
      id: '4',
      type: 'mascarpone',
      pricePerKilo: 340,
      color: 'white',
      pictureFile: 'https://cheese.com/media/img/cheese/mascarpone.jpg',
    },
    {
      id: '5',
      type: 'cheddar',
      pricePerKilo: 10,
      color: 'yellow',
      pictureFile: 'https://cheese.com/media/img/cheese/wiki/cheddar.jpg',
    },
];

describe("Cheeseria", () => {
  it("can render", async () => {
    expect.hasAssertions();

    global.fetch = jest.fn().mockImplementationOnce(() => {
    return new Promise((resolve, reject) => {
        resolve({
        ok: true,
        status: 200,
        json: () => {
            return fakeData;
        },
        });
    });
    });

    const { baseElement } = render(<Cheeseria/>)
    await waitFor(() =>
        expect(screen.getByTestId("gouda-test-id")).toHaveTextContent("Type: gouda")
    );
   
    // TODO: more tests to add      
    expect(baseElement).toMatchSnapshot();
      
  });
});
