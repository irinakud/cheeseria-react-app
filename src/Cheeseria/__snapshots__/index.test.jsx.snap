// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheeseria can render 1`] = `
<body>
  <div>
    <ul>
      <li
        data-testid="gouda-test-id"
      >
        <div>
          <h3>
            Type: 
            gouda
          </h3>
          <div>
            <img
              alt="gouda"
              src="https://cheese.com/media/img/cheese/GOUDA_SqYJjRh.jpg"
              style="height: 200px;"
            />
          </div>
          <div>
            Color: 
            yellow
          </div>
          <div>
            Price per kilo: 
            81
          </div>
        </div>
        <div>
          <div>
            Enter amount (in kg):
            <input />
          </div>
          <div>
            Total Price: 
            0
          </div>
        </div>
      </li>
      <li
        data-testid="mozzarella-test-id"
      >
        <div>
          <h3>
            Type: 
            mozzarella
          </h3>
          <div>
            <img
              alt="mozzarella"
              src="https://cheese.com/media/img/cheese/Mozzarella.jpg"
              style="height: 200px;"
            />
          </div>
          <div>
            Color: 
            white
          </div>
          <div>
            Price per kilo: 
            17
          </div>
        </div>
        <div>
          <div>
            Enter amount (in kg):
            <input />
          </div>
          <div>
            Total Price: 
            0
          </div>
        </div>
      </li>
      <li
        data-testid="brie-test-id"
      >
        <div>
          <h3>
            Type: 
            brie
          </h3>
          <div>
            <img
              alt="brie"
              src="https://cheese.com/media/img/cheese/Brie_PDCo3RG.jpg"
              style="height: 200px;"
            />
          </div>
          <div>
            Color: 
            yellow
          </div>
          <div>
            Price per kilo: 
            100
          </div>
        </div>
        <div>
          <div>
            Enter amount (in kg):
            <input />
          </div>
          <div>
            Total Price: 
            0
          </div>
        </div>
      </li>
      <li
        data-testid="mascarpone-test-id"
      >
        <div>
          <h3>
            Type: 
            mascarpone
          </h3>
          <div>
            <img
              alt="mascarpone"
              src="https://cheese.com/media/img/cheese/mascarpone.jpg"
              style="height: 200px;"
            />
          </div>
          <div>
            Color: 
            white
          </div>
          <div>
            Price per kilo: 
            340
          </div>
        </div>
        <div>
          <div>
            Enter amount (in kg):
            <input />
          </div>
          <div>
            Total Price: 
            0
          </div>
        </div>
      </li>
      <li
        data-testid="cheddar-test-id"
      >
        <div>
          <h3>
            Type: 
            cheddar
          </h3>
          <div>
            <img
              alt="cheddar"
              src="https://cheese.com/media/img/cheese/wiki/cheddar.jpg"
              style="height: 200px;"
            />
          </div>
          <div>
            Color: 
            yellow
          </div>
          <div>
            Price per kilo: 
            10
          </div>
        </div>
        <div>
          <div>
            Enter amount (in kg):
            <input />
          </div>
          <div>
            Total Price: 
            0
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;
